libraries/WiFi/examples/WiFiScan/WiFiScan.ino

```diff
@@ -19,30 +19,69 @@ void setup()
 
 void loop()
 {
-    Serial.println("scan start");
+    Serial.println("Scan start");
 
-    // WiFi.scanNetworks will return the number of networks found
+    // WiFi.scanNetworks will return the number of networks found.
     int n = WiFi.scanNetworks();
-    Serial.println("scan done");
+    Serial.println("Scan done");
     if (n == 0) {
         Serial.println("no networks found");
     } else {
         Serial.print(n);
         Serial.println(" networks found");
+        Serial.println("Nr | SSID            | RSSI | CH | Encryption");
         for (int i = 0; i < n; ++i) {
             // Print SSID and RSSI for each network found
-            Serial.print(i + 1);
-            Serial.print(": ");
-            Serial.print(WiFi.SSID(i));
-            Serial.print(" (");
-            Serial.print(WiFi.RSSI(i));
-            Serial.print(")");
-            Serial.println((WiFi.encryptionType(i) == WIFI_AUTH_OPEN)?" ":"*");
+            Serial.printf("%2d",i + 1);
+            Serial.print(" | ");
+            if (WiFi.SSID(i).length() < 15) {
+                Serial.printf("%-15s", WiFi.SSID(i).c_str());
+            } else {
+                Serial.print(WiFi.SSID(i));
+            }
+            Serial.print(" | ");
+            Serial.printf("%4d", WiFi.RSSI(i));
+            Serial.print(" | ");
+            Serial.printf("%2d", WiFi.channel(i));
+            Serial.print(" | ");
+            switch (WiFi.encryptionType(i))
+            {
+            case WIFI_AUTH_OPEN:
+                Serial.print("open");
+                break;
+            case WIFI_AUTH_WEP:
+                Serial.print("WEP");
+                break;
+            case WIFI_AUTH_WPA_PSK:
+                Serial.print("WPA");
+                break;
+            case WIFI_AUTH_WPA2_PSK:
+                Serial.print("WPA2");
+                break;
+            case WIFI_AUTH_WPA_WPA2_PSK:
+                Serial.print("WPA+WPA2");
+                break;
+            case WIFI_AUTH_WPA2_ENTERPRISE:
+                Serial.print("WPA2-EAP");
+                break;
+            case WIFI_AUTH_WPA3_PSK:
+                Serial.print("WPA3");
+                break;
+            case WIFI_AUTH_WPA2_WPA3_PSK:
+                Serial.print("WPA2+WPA3");
+                break;
+            case WIFI_AUTH_WAPI_PSK:
+                Serial.print("WAPI");
+                break;
+            default:
+                Serial.print("unknown");
+            }
+            Serial.println();
```

Comment:
Suggest adding after this line the following to release from memory the results of the last scan. (Though hard to confirm from the docs if this is required when using the methods above to extract the results of the scan. Unclear if those methods release memory as each scan item is pulled from the queue.)

`WiFi.scanDelete();      // delete the results from the scan`

Reply:
Would be wrong here.  Scan = complete scan result, so the loop would fail.

For educational purpose the call could be added, so people do not run into memory issues.